drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi_dma.c

```diff
@@ -22,14 +35,62 @@ struct spi_dma_stream {
 	uint32_t channel;
 	struct dma_config dma_cfg;
 	struct dma_block_config dma_blk_cfg;
-	bool chunk_done;
 };
 
 struct spi_nxp_dma_data {
 	struct spi_dma_stream dma_rx;
 	struct spi_dma_stream dma_tx;
+
+	uint32_t spi_transfer_state;
+	/* This DMA size is used in callback function for RX and TX context update.
+	 * because of old LPSPI IP limitation, RX complete depend on next TX DMA transfer start,
+	 * so TX and RX not always start at the same time while we can only calculate DMA transfer
+	 * size once and update the buffer pointers at the same time.
+	 * -1 means invalid DMA transfer size, need update.
+	 */
+	int32_t synchronize_dma_size;
 };
 
+/* Issue a TCR is requirement on some version of LPSPI or last RX DMA transfer will
+ * never end. It also depends on flag SPI_HOLD_ON_CS, if it is set, we will try to
+ * keep PCS asserted during transanctions.
+ * What this function does is: Check current SPI configuration and LPSPI version,
+ * if SPI_HOLD_ON_CS is required and this LPSPI version support "Keep PCS asserting
+ * during multi-transfers", do nothing.
+ * otherwise, issue a transimit command.
+ * On LPSPI main version 01: TCR is needed.
+ * On LPSPI main version 02: TCR is not needed.
+ * You can easily get the LPSPI main version number in register: Version ID (VERID)
+ * It is usally the 1st register in memory map.
+ * Below is a summary of what LPSPI version used for most of SoCs:
+ * LPSPI version 01: RT1170, RT10xx, Kinetis K serials
+ * LPSPI version 02: RT1180, MCXN, RT700, K32W, S32K3xx, MCXL10
+ */
+static void spi_mcux_issue_TCR(const struct device *dev)
+{
+	LPSPI_Type *base = (LPSPI_Type *)DEVICE_MMIO_NAMED_GET(dev, reg_base);
+	const struct spi_config *spi_cfg = DEV_DATA(dev)->ctx.config;
+	uint8_t major_ver = (base->VERID & 0xFF000000UL) >> 24;
+
+	if (major_ver > 1) {
+		/* Diable continuous command mode will issue a frame end at the end of
+		 * each transfer when SPI_HOLD_ON_CS is not set.
+		 */
+		if (!(spi_cfg->operation & SPI_HOLD_ON_CS)) {
```

Comment:
nit, can do if (major_ver > 1 && !(spi_cfg->operation & SPI_HOLD_ON_CS))

to reduce a level of indenting

Reply:
yes, it looks simpler but from the logic, it is less clear. 
not a big deal but I would like to make it easier to understand.

Reply:
I actually think it does make it easier to understand to combine them, because what you are actually checking for is saying "If we are version 1, and the flag isnt set, then ok". An if nested within an if is an "AND"